app/lib/features/chat_ng/widgets/events/text_message_event.dart

```diff
@@ -8,23 +11,47 @@ import 'package:flutter_matrix_html/text_parser.dart';
 class TextMessageEvent extends StatelessWidget {
   final String roomId;
   final MsgContent content;
-  final bool isUser;
-  final bool nextMessageGroup;
-  final bool wasEdited;
+  final Map<String, dynamic> metadata;
```

Comment:
why swap nicely typed specific fields that our compiler can check for us with a dynamic map of undefined stuff that then goes bad at runtime? I'd rather like to see _more_ typing and less...

Reply:
Yeah as stated above, will incorporate in a class model.